Sources/StreamVideoSwiftUI/Livestreaming/LivestreamPlayer.swift

```diff
@@ -64,74 +83,174 @@ public struct LivestreamPlayer<Factory: ViewFactory>: View {
+        _state = ObservedObject(wrappedValue: call.state)
+        self.joinPolicy = joinPolicy
+        self.showsLeaveCallButton = showsLeaveCallButton
+        self.onFullScreenStateChange = onFullScreenStateChange
+        call.updateParticipantsSorting(with: livestreamOrAudioRoomSortPreset)
+    }
+    
+    internal init(
+        viewFactory: Factory = DefaultViewFactory.shared,
+        call: Call,
+        countdown: TimeInterval = 0,
+        livestreamState: LivestreamState? = nil,
+        muted: Bool = false,
+        showParticipantCount: Bool = true,
+        joinPolicy: JoinPolicy = .auto,
+        showsLeaveCallButton: Bool = false,
+        onFullScreenStateChange: ((Bool) -> Void)? = nil
+    ) {
+        self.viewFactory = viewFactory
+        self.call = call
+        self.countdown = countdown
+        self.livestreamState = livestreamState ?? (call.state.backstage ? .backstage : .live)
+        self.muted = muted
+        self.showParticipantCount = showParticipantCount
+        _state = ObservedObject(wrappedValue: call.state)
         self.joinPolicy = joinPolicy
         self.showsLeaveCallButton = showsLeaveCallButton
         self.onFullScreenStateChange = onFullScreenStateChange
-        viewModel.call.updateParticipantsSorting(with: livestreamOrAudioRoomSortPreset)
+        call.updateParticipantsSorting(with: livestreamOrAudioRoomSortPreset)
     }
 
     public var body: some View {
         ZStack {
-            if viewModel.errorShown {
+            if livestreamState == .error {
                 errorView
-            } else if viewModel.loading {
+            } else if livestreamState == .joining {
                 loadingView
-            } else if state.backstage {
+            } else if livestreamState == .backstage {
                 notStartedView
             } else {
                 videoRenderer
+                livestreamControls
             }
-            livestreamControls
         }
         .onChange(of: state.participants, perform: { newValue in
-            if viewModel.muted && newValue.first?.track != nil {
-                viewModel.muteLivestreamOnJoin()
+            if muted && newValue.first?.track != nil {
+                muteLivestreamOnJoin()
             }
         })
+        .onChange(of: call.state.backstage) { _ in
+            livestreamState = call.state.backstage ? .backstage : .live
+            if let startsAt = state.startsAt, livestreamState == .backstage && timer == nil {
+                timer = Timer.scheduledTimer(withTimeInterval: 1, repeats: true) { _ in
+                    Task { @MainActor in
+                        countdown = startsAt.timeIntervalSinceNow
+                        if countdown <= 0 {
+                            stopTimer()
+                            countdown = 0
+                        }
+                    }
+                }
+            }
+            if livestreamState == .live {
+                stopTimer()
+            }
+        }
```

Comment:
```suggestion
        .onChange(of: call.state.backstage) { _ in
            livestreamState = call.state.backstage ? .backstage : .live
            if
                let startsAt = state.startsAt,
                livestreamState == .backstage,
                timerCancellable == nil
            {
                timerCancellable = Timer
                    .publish(every: 1, on: .main, in: .default)
                    .autoconnect()
                    .sinkTask { @MainActor _ in
                        countdown = startsAt.timeIntervalSinceNow
                        if countdown <= 0 {
                            stopTimer()
                            countdown = 0
                        }
                    }
            } else if livestreamState == .live {
                stopTimer()
            }
        }
```
Can we switch the Combine version of timer?

Reply:
sure